stdlib/2/pydoc.pyi

```diff
@@ -0,0 +1,176 @@
+    def bigsection(self, title: str, *args) -> str: ...
+    def preformat(self, text: str) -> str: ...
+    def multicolumn(self, list: List[Any], format: Callable[[Any], str], cols: int = ...) -> str: ...
+    def grey(self, text: str) -> str: ...
+    def namelink(self, name: str, *dicts: Tuple[MutableMapping[str, str]]) -> str: ...
+    def classlink(self, object: object, modname: str) -> str: ...
+    def modulelink(self, object: object) -> str: ...
+    def modpkglink(self, data: Tuple[str, str, bool, bool]) -> str: ...
+    def markup(self, text: str, escape: Optional[Callable[[str], str]] = ..., funcs: Mapping[str, str] = ..., classes: Mapping[str, str] = ..., methods: Mapping[str, str] = ...) -> str: ...
+    def formattree(self, tree: List[Union[Tuple[type, Tuple[type, ...]], list]], modname: str, parent: Optional[type] = ...) -> str: ...
+    def docmodule(self, object: object, name: Optional[str] = ..., mod: Optional[str] = ..., *ignored) -> str: ...
+    def docclass(self, object: object, name: Optional[str] = ..., mod: Optional[str] = ..., funcs: Mapping[str, str] = ..., classes: Mapping[str, str] = ..., *ignored) -> str: ...
+    def formatvalue(self, object: object) -> str: ...
+    def docroutine(self, object: object, name: Optional[str] = ..., mod: Optional[str] = ..., funcs: Mapping[str, str] = ..., classes: Mapping[str, str] = ..., methods: Mapping[str, str] = ..., cl: Optional[type] = ..., *ignored) -> str: ...
+    def docproperty(self, object: object, name: Optional[str] = ..., mod: Optional[str] = ..., cl: Optional[Any] = ..., *ignored) -> str: ...
+    def docother(self, object: object, name: Optional[str] = ..., mod: Optional[Any] = ..., *ignored) -> str: ...
+    def docdata(self, object: object, name: Optional[str] = ..., mod: Optional[Any] = ..., cl: Optional[Any] = ..., *ignored) -> str: ...
+    def index(self, dir: str, shadowed: Optional[MutableMapping[str, bool]] = ...) -> str: ...
+
+class TextRepr(Repr):
+    maxlist = ...  # type: int
+    maxdict = ...  # type: int
+    maxstring = ...  # type: int
+    def __init__(self) -> None: ...
+    def repr1(self, x: object, level: complex) -> str: ...
+    def repr_string(self, x: str, level: complex) -> str: ...
+    def repr_str(self, x: str, level: complex) -> str: ...
+    def repr_instance(self, x: object, level: complex) -> str: ...
+
+class TextDoc(Doc):
+    def repr(self, object: object) -> str: ...
+    def bold(self, text: str) -> str: ...
+    def indent(self, text: str, prefix: str = ...) -> str: ...
+    def section(self, title: str, contents: str) -> str: ...
+    def formattree(self, tree: List[Union[Tuple[type, Tuple[type, ...]], list]], modname: str, parent: Optional[type] = ..., prefix: str = ...) -> str: ...
+    def docmodule(self, object: object, name: Optional[str] = ..., mod: Optional[Any] = ..., *ignored) -> str: ...
```

Comment:
Oh, I see. Because `Doc.docmodule` has `*args`, it can take any number of arguments, right? So `HTMLDoc.docmodule` must _also_ be able to take any number of arguments, even though the actual code has a fixed number of arguments. It's a hack around the subclasses' methods having different parameter counts than the base class.